### ISSUE TITLE
Balance not calculated correctly

### ISSUE THREAD

**Issue by whalley** [moneymanagerex/moneymanagerex#3509]
MMEX version:
 - [X] 1.6.0 Alpha 1
 - [ ] 1.3.6

Operating System:
 - [ ] Windows  
 - [X] Mac OSX
 - [ ] Linux 

# Description of the bug

Balance being calculated incorrectly. See below. Very odd.

If the deposit is **23.51** or **23.53** the balance is calculated correctly (almost seems to be some form of rounding error).

# Reproduction

Is the bug reproducible? 
 - [X] Always 
 - [ ] Randomly 
 - [ ] Happened only once

![CleanShot 2021-05-25 at 20 45 17](https://user-images.githubusercontent.com/487345/119559383-7d50e980-bd9a-11eb-81f3-11793477c907.png)

**Comment by vomikan** [comment]
![image](https://user-images.githubusercontent.com/6836805/119562984-a8920380-bdaf-11eb-855d-df9f1e175c51.png)

**Comment by whalley** [comment]
Yikes, my Mac can't count then :-)    

Hmmmm... Think I'll see if I can create a test db with this in.

**Comment by vomikan** [comment]
This is probably due to the rejection of Intel processors ))

**Comment by vomikan** [comment]
A similar problem. The calculations are done correctly, but the representation is not.
https://github.com/moneymanagerex/moneymanagerex/issues/3494

**Comment by whalley** [comment]
[test3509.mmb.zip](https://github.com/moneymanagerex/moneymanagerex/files/6542504/test3509.mmb.zip)

Does this show a balance on the "ERROR ACCOUNT" of £141.00 or £142.00 ..... for me it shows £141.00

**Comment by whalley** [comment]
> A similar problem. The calculations are done correctly, but the representation is not.
> #3494

Yes did see that...

**Comment by vomikan** [comment]
![image](https://user-images.githubusercontent.com/6836805/119573615-30323f00-bdbd-11eb-89e2-a06883823522.png)

**Comment by whalley** [comment]
Good, so it is not just me then!  Thanks for checking, one to try to get to the bottom of tomorrow.

**Comment by tactilis** [comment]
On Windows:
-£141.00 using 1.6.0 Alpha.1
-£142.00 using 1.5.1

**Comment by vomikan** [comment]
![image](https://user-images.githubusercontent.com/6836805/119574689-894ea280-bdbe-11eb-98c8-fbe0503577ab.png)

**Comment by whalley** [comment]
Yep that `abs(static_cast<int>(value))` is the culprit. Need to round to precision before we do anything. Can't think of a more elagant way of doing it without converting to string and then back to to int/float for the `{:L} `localisation.

**Comment by vomikan** [comment]
#2473
